File: challenge.js
// Year Check
// tip use Date object

// Given a year and month determine the number of years (two decimal accuracy) between
// the given time and now

// advanced challenge
// give day
// and go in the future



var now = new Date()
var readlineSync = require('readline-sync');
let currentYear = now.getFullYear()
let currentMonth = now.getMonth() + 1;

let userYear = parseInt(readlineSync.question("Enter a year "));
let userMonth = parseInt(readlineSync.question("Enter a month xx format "));

console.log(userYear);
console.log(userMonth);

if (!Number.isInteger(userMonth)){
    console.log("Month but be in number format")
    return 1;
}

if (userYear > currentYear){
    let futureYearlyDifference = userYear - currentYear;
    let futureMonthDifference = Math.abs((userMonth - currentMonth) / 12)
    let futureTotal = futureYearlyDifference + futureMonthDifference;

    console.log(`${futureTotal.toFixed(2)} years ahead`)
    
} else {
    let yearlyDifference = currentYear - userYear;
    let monthlyDifference = Math.abs(((currentMonth - userMonth) / 12)) 
    let total = yearlyDifference + monthlyDifference;
    
    console.log(`${total.toFixed(2)} years`)
}




// console.log(`${yearlyDifference}${monthyDifference}`)

File: challenge.c
#include <stdio.h>
#include <stdlib.h>
#include <time.h>
#include <math.h>
#include <ctype.h>

int main(void)
{
    /* Given a year and month determine the number of years (two decimal accuracy) between
    the given time and now */

    int temp1, currentMonth, currentYear, userYear, yearDifference, futureYearDifference;
    float temp2, userMonth, monthDifference, futureMonthDifference, total, futureTotal;

    //time_t is an arithmitic type for representing time
    time_t now;

    time(&now);
    struct tm *current = localtime(&now); // use the address of now to fill the structure with values representing local time
    currentMonth = current->tm_mon + 1; // month from (0 to 11) + 1 so that currentMonth would be from 1 - 12
    currentYear = current->tm_year + 1900; // years from 1900 than add 1900 to get currentYear
    
    printf("Enter Year xxxx format: \n");
    scanf("%d", &temp1);

    printf("Enter Month xx format: \n");
    scanf("%f", &temp2);

    userYear = temp1; 
    userMonth = temp2;

    if (userYear > currentYear)
    {
        futureYearDifference = userYear - currentYear;
        futureMonthDifference = fabs((userMonth - currentMonth) / 12);
        futureTotal = futureYearDifference + futureMonthDifference;
        printf("%.2f years ahead.\n", futureTotal);
    }

    else
    {
        yearDifference = currentYear - userYear;
        monthDifference = fabs((currentMonth - userMonth) / 12);
        total = yearDifference + monthDifference;
        printf("%.2f year difference.\n", total);
    }

    
   




    
    
    // int currentYear;
    // printf()

    return 0;

}